packages/view/src/components/Statistics/AuthorBarChart/AuthorBarChart.tsx

```diff
@@ -0,0 +1,123 @@
+import type { ChangeEvent } from "react";
+import { useRef, useEffect, useState } from "react";
+import * as d3 from "d3";
+
+import type { ClusterNode } from "types";
+
+import type { AuthorDataType, MetricType } from "./AuthorBarChart.type";
+import { getDataByAuthor, sortDataByName } from "./AuthorBarChart.util";
+import { DIMENSIONS, METRIC_TYPE } from "./AuthorBarChart.const";
+
+import "./AuthorBarChart.scss";
+
+type AuthorBarChartProps = {
+  data: ClusterNode[];
+};
+
+const AuthorBarChart = ({ data: rawData }: AuthorBarChartProps) => {
+  const svgRef = useRef(null);
+  const [metric, setMetric] = useState<MetricType>(METRIC_TYPE[0]);
+
+  const authorData = getDataByAuthor(rawData);
+  const totalMetricValues = authorData.reduce(
+    (acc, item) => acc + item[metric],
+    0
+  );
+
+  useEffect(() => {
+    const data = authorData.sort((a, b) => {
+      if (a[metric] === b[metric]) {
+        return sortDataByName(a.name, b.name);
+      }
+      return a[metric] - b[metric];
+    });
```

Comment:
데이터 가공은 useEffect 밖에서 끝내는 것은 어떨까요!? 

```suggestion
  const authorData = getDataByAuthor(rawData);

  const data = authorData.sort((a, b) => {
    if (a[metric] === b[metric]) {
      return sortDataByName(a.name, b.name);
    }
    return a[metric] - b[metric];
  });

  const totalMetricValues = authorData.reduce(
    (acc, item) => acc + item[metric],
    0
  );

```

Reply:
eslint의 옛날 버전에서는 react-hooks/exhaustive-deps rule에 저 data 같은 애들이 전부 걸렸는데,
최근 버전에는 스마트하게 동작하는 것 같네요? (제가 넘 옛날 버전을 쓰고 있었나...)
https://stackoverflow.com/questions/58624200/react-hook-useeffect-has-a-missing-dependency-dispatch/58624276#58624276

밖으로 빼는 것도 괜찮겠습니다만, 다른 데서 안 쓰는 func이라면 cohesion 측면에서는 블락 하나라고 생각하고 뭉치는 것도 뭐 괜찮을 것 같기도 하네요 ㅋ

Reply:
밖으로 빼는 게 더 깔끔해보이긴 하는데 멘토님 말씀처럼 `react-hooks/exhaustive-deps`룰에 걸려 depth가 엄청 길어집니다😢
ex) `[rawData, metric, authorData, totalMetricValues, data]);`
이 부분 일단 useEffect 안에 넣어두는 것으로 진행하겠습니다.

Reply:
exhaustive deps rule이 적용 안되어 있었군요 @__@. 저 룰 있어야하고 deps에 문제가 되면 넣어놓는게 맞지용. 알겠습니다 ㅎ_ㅎ